src/repl/scala/tools/nsc/interpreter/package.scala

```diff
@@ -88,8 +88,12 @@ package object interpreter extends ReplConfig with ReplStrings {
           }
       }
 
-      if (filtered.isEmpty)
+      if (filtered.isEmpty) {
+        if (global.settings.nopredef || global.settings.noimports) {
+          return "No implicits have been imported."
+        }
         return "No implicits have been imported other than those in Predef."
+      }
```

Comment:
Always desirable to get rid of returns. The code is confusing anyway b/c side-effects. I don't know why it doesn't return a string to output? I don't want to know. If you really think baby-step of if-else at bottom is too confusing, put the foreach behind a `def printStuff()` so it's `if (stuff.isEmpty) "str" else { printStuff(); "" }` or similar.